js/transitions.js

```diff
@@ -21,101 +25,70 @@ export async function transitionsEnded($element, property = null) {
         .off('transitioncancel transitionend', onEnd);
       resolve(didTransition);
     };
-    if (!willTransition($element, property)) return resolve(false);
     $element.on('transitionrun', onRun);
     $element.on('transitioncancel transitionend', onEnd);
     const resolveInterval = setTimeout(() => {
-      const element = $element[0];
-      const id = element.name || element.id || element.className;
-      logging.warn(`transition could/did not run for '${id}', forcing resolution`);
+      logging.warn('transition could/did not run forcing resolution', $element[0]);
       if (!hasActiveTransition($element, property)) done(false);
-    }, getResolutionDuration($element, property));
+    }, timeoutDuration);
   });
 }
 
 /**
- * Returns all `CSSTransitions` currently affecting an element.
+ * Returns the longest end time of the configured transitions
  * @param {jQuery} $element
- * @returns {Array<CSSTransition>}
+ * @param {string} [property=null]
+ * @returns {number}
  */
-export function getActiveTransitions($element) {
-  let element = $element;
-  if (element instanceof $) element = element[0];
-  return element.getAnimations().filter(animation => animation instanceof CSSTransition);
-}
-
-/**
- * Returns a transition object by evaluating the CSS properties of an element.
- * @param {jQuery} $element
- * @returns {Object}
- */
-export function getTransitionsFromCSS($element) {
+export function getTransitionsLongestEndTime($element, property = null) {
   const properties = $element.css('transition-property').split(',').map(property => property.trim());
   const durations = $element.css('transition-duration').split(',').map(duration => parseFloat(duration));
   const delays = $element.css('transition-delay').split(',').map(delay => parseFloat(delay));
-  return properties.map((transitionProperty, index) => {
+  const endTimes = properties.reduce((result, transitionProperty, index) => {
     // cycle through populated timings as per spec
     const duration = durations[index % durations.length];
     const delay = delays[index % delays.length];
-    const timeToEnd = duration + delay;
-    return {
-      transitionProperty,
-      timeToEnd
-    }
-  });
+    const endTime = duration + delay;
+    result[transitionProperty] = endTime;
+    return result;
+  }, {});
+  const isAllProperty = property === 'all' || Boolean(endTimes.all);
+  const longestEndTime = (property && !isAllProperty)
+    ? endTimes[property]
+    : Object.values(endTimes).sort((a, b) => a.timeToEnd - b.timeToEnd).pop();
```

Comment:
`timeToEnd` no longer exists, so this will just return the last item in the Array without sorting. As the object now uses the property name as the key, this could be changed to:

```suggestion
    : Math.max(...Object.values(endTimes));
```

Reply:
> It's possible to log the element instead of having to derive an id or name.

I thought printing the whole element would be too verbose, but as this is now in a second argument I guess it adds more detail and can be expanded to reveal the element as required.

> getActiveTransitions is only used inside willTransition, there is probably no need to split this into two functions as the code never repeats.

This makes sense for the code within `transitionsEnded`, but I had purposefully split these up should they need be useful by other plugins in future. Fine for these to be merged though as probably unlikley to be needed.

Reply:
`getActiveTransitions` is pretty much just the native `getAnimations` with the `CSSTransition` class check, so not much hassle to add late or recreate.

Reply:
> `timeToEnd` no longer exists, so this will just return the last item in the Array without sorting. As the object now uses the property name as the key, this could be changed to:

Ominous bit of code to read out loud.